src/content/developers/docs/data-structures-and-encoding/rlp/index.md

```diff
@@ -20,21 +20,25 @@ The RLP encoding function takes in an item. An item is defined as follows：
 
 - a string (i.e. byte array) is an item
 - a list of items is an item
+- a positive integer is an item
 
 For example, all of the following are items:
 
 - an empty string;
 - the string containing the word "cat";
 - a list containing any number of strings;
 - and a more complex data structures like `["cat", ["puppy", "cow"], "horse", [[]], "pig", [""], "sheep"]`.
+- the number `100`
 
-Note that in the context of the rest of this page, 'string' means "a certain number of bytes of binary data"; no special encodings are used, and no knowledge about the content of the strings is implied.
+Note that in the context of the rest of this page, 'string' means "a certain number of bytes of binary data"; no special encodings are used, and no knowledge about the content of the strings is implied (except as required by the rule against non-minimal positive integers).
 
 RLP encoding is defined as follows:
 
+- For a positive integer, it is converted to the the shortest byte array whose big-endian interpretation is the integer, and then encoded as a string according to the rules below.
 - For a single byte whose value is in the `[0x00, 0x7f]` (decimal `[0, 127]`) range, that byte is its own RLP encoding.
 - Otherwise, if a string is 0-55 bytes long, the RLP encoding consists of a single byte with value **0x80** (dec. 128) plus the length of the string followed by the string. The range of the first byte is thus `[0x80, 0xb7]` (dec. `[128, 183]`).
 - If a string is more than 55 bytes long, the RLP encoding consists of a single byte with value **0xb7** (dec. 183) plus the length in bytes of the length of the string in binary form, followed by the length of the string, followed by the string. For example, a 1024 byte long string would be encoded as `\xb9\x04\x00` (dec. `185, 4, 0`) followed by the string. Here, `0xb9` (183 + 2 = 185) as the first byte, followed by the 2 bytes `0x0400` (dec. 1024) that denote the length of the actual string. The range of the first byte is thus `[0xb8, 0xbf]` (dec. `[184, 191]`).
```

Comment:
> length of the string in binary form

This is also ambiguous and should specify endianess and leading-zeros treatment. Same with length of the list below.

Reply:
Yeah, there are a number of other fixes that could be made, but I wanted to limit this PR to clarifying the specific weirdness wrt RLP definition of positive integer encoding leaking into higher-level protocols 😮‍💨